pandas/core/reshape/reshape.py

```diff
@@ -1089,6 +1089,198 @@ def get_empty_frame(data) -> DataFrame:
+        )
+
+    if data[subset].isna().any().any():
+        raise ValueError(
+            f"Dummy DataFrame contains NA value in column: "
+            f"'{data.columns[data.isna().any().argmax()]}'"
+        )
+
+    # index data with a list of all columns that are dummies
+    try:
+        data_to_decode = data[subset].astype("boolean")
+    except TypeError:
+        raise TypeError("Passed DataFrame contains non-dummy data")
+
+    # get separator for each prefix and lists to slice data for each prefix
+    if sep is None:
+        variables_slice = {"categories": subset}
+    elif isinstance(sep, dict):
+        variables_slice = {prefix: [] for prefix in sep}
+        for col in data_to_decode.columns:
+            for prefix in sep:
+                if prefix in col:
+                    variables_slice[prefix].append(col)
+    else:
+        sep_for_prefix = {}
+        variables_slice = {}
+        for col in data_to_decode.columns:
+            ps = [ps for ps in sep if ps in col]
+            if len(ps) == 0:
+                raise ValueError(
+                    f"Separator not specified for all columns; "
+                    f"First instance column: '{col}'"
+                )
+            prefix = col.split(ps[0])[0]
+            if prefix not in sep_for_prefix:
+                sep_for_prefix[prefix] = ps[0]
+            if prefix not in variables_slice:
+                variables_slice[prefix] = [col]
+            else:
+                variables_slice[prefix].append(col)
+        sep = sep_for_prefix
+
+    # validate number of dropped_first
+    def check_len(item, name) -> None:
+        if not len(item) == len(variables_slice):
+            len_msg = (
+                f"Length of '{name}' ({len(item)}) did not match the "
+                "length of the columns being encoded "
+                f"({len(variables_slice)})"
+            )
+            raise ValueError(len_msg)
+
+    if dropped_first:
+        if isinstance(dropped_first, dict):
+            check_len(dropped_first, "dropped_first")
+        elif isinstance(dropped_first, list):
+            check_len(dropped_first, "dropped_first")
+            dropped_first = dict(zip(variables_slice, dropped_first))
+        elif isinstance(dropped_first, str):
+            dropped_first = dict(
+                zip(variables_slice, [dropped_first] * len(variables_slice))
+            )
+        else:
+            raise TypeError(
+                f"Expected 'dropped_first' to be of type 'str', 'list', or 'dict'; "
+                f"Received 'dropped_first' of type: {type(dropped_first).__name__}"
+            )
+
+    cat_data = {}
+    for prefix, prefix_slice in variables_slice.items():
+        if sep is None:
+            cats = subset.copy()
+        else:
+            cats = [col[len(prefix + sep[prefix]) :] for col in prefix_slice]
+        assigned = data_to_decode[prefix_slice].sum(axis=1)
+        if any(assigned > 1):
+            raise ValueError(
```

Comment:
Couldn't you check this much earlier with a row sum after the conversion to boolean, e.g. , if (data_to_decode.sum(1) > 1).any()?

Reply:
Yes, this is still left from when I mentioned `row` as well as the `prefix` in the error message (which I removed together with `to_series`) which did look weird when there was actual prefix. I will move this check up, thank you for the hint +1